components/courses/app/views/admin/courses/progress/index.slim

```diff
@@ -26,8 +26,5 @@
           td = homework_mark(current_lecture, student)
           - if authenticate_lecturer!
             td
-              = set_mark_button(current_season, current_lecture, student, -1)
-              = set_mark_button(current_season, current_lecture, student, 0)
-              = set_mark_button(current_season, current_lecture, student, 1)
-              = set_mark_button(current_season, current_lecture, student, 1.5)
-              = set_mark_button(current_season, current_lecture, student, 2)
+              - ::Courses::Progress::ALLOWED_MARKS.drop(1).each do |mark|
```

Comment:
Instead of explicitly dropping the first element of an array, consider using `Array#compact` - a method that removes all nil elements. This way, if someone later changes the ALLOWED_MARKS array in such way that nil is not the first element, the code will continue to work well.

Reply:
Thanks, i have changed it.